socket/sandboxes/heka/output/heka_tcp.lua

```diff
@@ -59,26 +59,25 @@ end
 
 local client, err = create_client()
 
-local function send_message(msg, i)
-    local len, err, i = client:send(msg, i)
-    if not len then
-        if err == "timeout" or err == "closed" then
-            client:close()
-            client = nil
-            return -3, err
+local function send_message(msg)
+    local i = 1
+    for retry = 1, 3 do
+        local len, err, i = client:send(msg, i)
+        if len then
+            return 0
```

Comment:
http://w3.impa.br/~diego/software/luasocket/tcp.html#send but they are clearly not 100% complete/accurate